### PR TITLE
feat(share): implements sync CascadeGetter

### PR DESCRIPTION
**Pull request by Wondertan**
An alternative version of #1581.

### CODE REVIEW

File: share/getters/cascade.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+		if err != nil {
+			// we do not set the actual errors to the description, as they were already recorded
+			span.SetStatus(codes.Error, "all getters failed")
+			return
+		}
+		span.SetStatus(codes.Ok, "")
+	}()
+
+	for i, getter := range getters {
+		span.AddEvent("getter launched", trace.WithAttributes(attribute.Int("getter_id", i)))
```
Comment (vgonkivs) [inline share/getters/cascade.go:120]: can we name it `getter_index`. IMO, `id` is something unique.

File: share/getters/cascade.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		ctx, cancel := context.WithTimeout(ctx, interval)
+		val, err := get(ctx, getter)
+		cancel()
+		if err == nil {
+			return val, nil
+		}
+
+		// TODO(@Wondertan): migrate to errors.Join once Go1.20 is out!
+		err = multierr.Append(err, err)
+		span.RecordError(err, trace.WithAttributes(attribute.Int("getter_id", i)))
```
Comment (vgonkivs) [inline share/getters/cascade.go:130]: The same here.

File: share/getters/cascade.go [hunk 3]
```diff
@@ -3,15 +3,17 @@ package getters
...
-	"github.com/celestiaorg/rsmt2d"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/multierr"
+
+	"github.com/celestiaorg/celestia-node/libs/utils"
+	"github.com/celestiaorg/celestia-node/share"
+
+	"github.com/celestiaorg/nmt/namespace"
+	"github.com/celestiaorg/rsmt2d"
```
Comment (Wondertan) [inline share/getters/cascade.go:16]: Order

File: share/getters/cascade.go (hunk 1 above)
Comment (Wondertan) [reply share/getters/cascade.go:114]: can be idx

**Review by distractedm1nd — DISMISSED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
lfg if @Wondertan likes

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**